hack/benchmark/internal/operation/search.go

```diff
@@ -167,12 +167,14 @@ func (o *operation) StreamSearchByID(ctx context.Context, b *testing.B, maxIdNum
 		}()
 
 		for i := 0; i < b.N; i++ {
-			err = sc.Send(&payload.Search_IDRequest{
-				Id:     strconv.Itoa(i % maxIdNum),
-				Config: cfg,
-			})
-			if err != nil {
-				b.Fatal(err)
+			if sc != nil {
```

Comment:
I think it should be better.

```go
...
if sc != nil {
    for i := 0; i < b.N; i++ {
...
```

Reply:
Since sc is also used inside the goroutine, I think check sc in all loops just to be safe.
But I'll update the source code for other reasons.

Reply:
pushed, please review it again.